src/ServicePulse.Host.Tests/tests/js/services/services.monitoring.spec.js

```diff
@@ -42,58 +42,55 @@
         $httpBackend.whenGET('http://localhost:33333/api/recoverability/endpoints').respond([]);
 
         var monitoredEndpoints;
-        monitoringService.getData().then(function (response) {
-            monitoredEndpoints = response.data;
+        monitoringService.getEndpoints().subscribe(function (response) {
+            monitoredEndpoints = response;
+            
+            expect(monitoredEndpoints.Name).toEqual("Samples.Metrics.Tracing.Endpoint");
+            expect(monitoredEndpoints.Count).not.toBeDefined();
+            expect(monitoredEndpoints.Data.Timestamps).toEqual([]);
+            expect(monitoredEndpoints.Data.CriticalTime).toEqual([]);
+            expect(monitoredEndpoints.Data.ProcessingTime).toEqual([]);
         });
         $httpBackend.flush();
-
-        expect(monitoredEndpoints["NServiceBus.Endpoints"].length).toEqual(1);
-        expect(monitoredEndpoints["NServiceBus.Endpoints"][0].Name).toEqual("Samples.Metrics.Tracing.Endpoint");
-        expect(monitoredEndpoints["NServiceBus.Endpoints"][0].Errors).not.toBeDefined();
-        expect(monitoredEndpoints["NServiceBus.Endpoints"][0].Data.Timestamps).toEqual([]);
-        expect(monitoredEndpoints["NServiceBus.Endpoints"][0].Data.CriticalTime).toEqual([]);
-        expect(monitoredEndpoints["NServiceBus.Endpoints"][0].Data.ProcessingTime).toEqual([]);
     });
 
-    it('should return filled collection when sc returns matching rows', function () {
+    it('should return added collections when sc returns matching rows', function () {
         $httpBackend.whenGET('http://localhost:1234/diagrams/data').respond(monitoredEndpointWithData);
         $httpBackend.whenGET('http://localhost:33333/api/recoverability/endpoints').respond(scEndpointWithMatchingData);
 
         var monitoredEndpoints;
-        monitoringService.getData().then(function (response) {
-            monitoredEndpoints = response.data;
+        monitoringService.getEndpoints().subscribe(function (response) {
+            monitoredEndpoints = response;
+            if (monitoredEndpoints.IsFromSC) {
```

Comment:
seeing this logic https://github.com/Particular/ServicePulse/pull/450/files#diff-db4ff24b7b6be144aa01f62db6bf3a2bR64 I'm even more convinced that we should not treat it as a single stream of data